unit-tests/src/test/java/org/eclipse/collections/impl/list/immutable/ImmutableArrayListTest.java

```diff
@@ -186,11 +186,17 @@ public void get()
         Verify.assertThrows(ArrayIndexOutOfBoundsException.class, () -> list.get(-1));
     }
 
-    @Test(expected = IllegalStateException.class)
     @Override
     public void iteratorRemove()
     {
-        this.classUnderTest().iterator().remove();
+        try
+        {
+            this.classUnderTest().iterator().remove();
```

Comment:
You need to call Assert.fail() in the try block to assert that an exception in fact gets thrown.